src/Tools/ExternalAccess/FSharp/Internal/Classification/FSharpClassificationService.cs

```diff
@@ -27,27 +29,33 @@ public FSharpClassificationService(IFSharpClassificationService service)
             _service = service;
         }
 
-        public void AddLexicalClassifications(SourceText text, TextSpan textSpan, List<ClassifiedSpan> result, CancellationToken cancellationToken)
+        public void AddLexicalClassifications(SourceText text, TextSpan textSpan, ArrayBuilder<ClassifiedSpan> result, CancellationToken cancellationToken)
         {
-            _service.AddLexicalClassifications(text, textSpan, result, cancellationToken);
+            using var _ = s_listPool.GetPooledObject(out var list);
+            _service.AddLexicalClassifications(text, textSpan, list, cancellationToken);
```

Comment:
Ideally F# should update their interface to take in an arraybuilder right?   Just want to check that they know

Reply:
Yup.  But we leave that up to them (though it's totally fine for us to notify them).  Effectively, they can choose to augment their api and do whatever they need, without affecting us.  it's the benefit of the EA pattern here.